test/unit/modules/blocks.js

```diff
@@ -9,6 +9,112 @@ var sinon = require('sinon');
 var modulesLoader = require('../../common/initModule').modulesLoader;
 
 describe('blocks', function () {
+	
+	describe('Blocks', function () {
+		
+		describe('library', function () {
+			
+			it('should assign logger');
+		});
+		
+		describe('this.submodules', function () {
+			
+			it('should assign api');
+			
+			it('should assign verify');
+			
+			it('should assign process');
+			
+			it('should assign utils');
+			
+			it('should assign chain');
+		});
+		
+		it('should set this.shared to api submodule');
+		
+		it('should set this.verify to verify submodule');
+		
+		it('should set this.process to process submodule');
+		
+		it('should set this.utils to utils submodule');
+		
+		it('should set this.chain to chain submodule');
+		
+		it('should set self to this');
+		
+		it('should call this.submodules.chain.saveGenesisBlock');
+		
+		describe('callback for this.submodules.chain.saveGenesisBlock', function () {
+			
+			it('should call callback with error');
+			
+			it('should call callback with self');
+		});
+	});
+	
+	describe('lastBlock', function () {
+		
+		it('should assign get');
+		
+		it('should assign set');
+		
+		it('should assign isFresh');
+	});
+	
+	describe('lastReciept', function () {
+		
+		it('should assign get');
+		
+		it('should assign update');
+		
+		it('should assign isStale');		
+	});
+	
+	describe('isActive', function () {
+		
+		it('should assign get');
+		
+		it('should assign set');			
+	});
+	
+	describe('isCleaning', function () {
+		
+		it('should assign get');			
+	});
+	
+	describe('onBind', function () {
+		
+		it('should set __private.loaded to true');
+	});
+	
+	describe('cleanup', function () {
+		
+		it('should set __private.loaded to false');
+		
+		it('should set __private.cleanup to true');
+		
+		describe('when __private.isActive is false', function () {
+			
+			it('should call callback');
+		});
+		
+		describe('when __private.isActive is true', function () {
+			
+			it('should recheck if __private.isActive is false every 10 seconds');
```

Comment:
I can see some difficulties with implementing test with 'every 10 seconds' keyword
```
describe('when __private.isActive = true')
   describe('after 10 seconds')
      it('should call logger...')
   describe('after 100 seconds')
      it('should call logger... 10 times')
```
what do you think?

Reply:
prefer to describe each in the way you described.